tsdb/block.go

```diff
@@ -107,6 +107,11 @@ type IndexReader interface {
 	Close() error
 }
 
+type ExtendedIndexReader interface {
+	IndexReader
+	PostingsForMatchers(ctx context.Context, ms ...*labels.Matcher) (index.Postings, error)
```

Comment:
Have you considered passing the wrapper to the original PostingsForMatcher call?  This would change the original interface but there are only 5 implementations outside of tests and that way you'd save the extra interface.

If the goal is to check if the postingformatchers call exists in a cache just by passing the wrapper as an argument you could do this. Normal prometheus will never have a wrapper and then won't use it but downstream projects could just modify how they call PostingForMatchers.